code/modules/mapping/mapping_helpers.dm

```diff
@@ -216,6 +216,16 @@
 	else
 		airlock.abandoned = TRUE
 
+/obj/effect/mapping_helpers/airlock/welded
+	name = "airlock welded helper"
+	icon_state = "airlock_welded"
+
+/obj/effect/mapping_helpers/airlock/welded/payload(obj/machinery/door/airlock/airlock)
+	if(airlock.welded)
+		log_mapping("[src] at [AREACOORD(src)] tried to make [airlock] welded but it's already welded closed!")
+	else
+		airlock.welded = TRUE
```

Comment:
does it show up as welded in-game? might have to do `update_appearance()` on the `airlock`

Reply:
It's in LateInitialize() on the parent. Also I tested it.